File: Python/2D_CNN_V2.py
%config Completer.use_jedi = False
import keras
import numpy as np
import zipfile as zip
import pandas as pd
import tensorflow as tf
import os
import re

#Set Working Directory
os.chdir(re.sub(pattern='Python', repl='', string=os.getcwd()))

#Load Train DF
BikeDF=pd.read_table('Python/Temp_X.txt')
BikeDF.shape
BikeDF.drop(axis=1, labels=['casual','registered'], inplace=True)
BikeDF.shape

#Load dependent var
BikeDF_Y=pd.read_table('Python/Temp_Y.txt')
BikeDF_Y.shape
#
TrainDF=np.array(BikeDF)
TrainDF=TrainDF.reshape((-1,3*2,4))
TrainDF_Y=np.array(BikeDF_Y)

TrainDF.shape
TrainDF_Y.shape


###########
model=keras.Sequential()
model.add(keras.layers.Conv1D(filters=32, kernel_size=1, activation='relu'))
model.add(keras.layers.MaxPooling1D(pool_size=1))
model.add(keras.layers.Conv1D(filters=16, kernel_size=1, activation='relu'))
model.add(keras.layers.MaxPooling1D(pool_size=1))
model.add(keras.layers.Flatten())
model.add(keras.layers.Dense(units=100, activation='relu'))
model.add(keras.layers.Dense(units=1))


#Compile Model
model.compile(optimizer='adam', loss='mse', metrics=['accuracy'])

#Fit model to Data
model.fit(TrainDF, TrainDF_Y, epochs=100, batch_size=16)

File: R/2D_CNN_V2.R
rm(list=ls())


#Library
library(tidyverse)
library(plyr)
library(keras)
library(reticulate)
#
use_python("C:/Program Files/Python37/")
np=import("numpy", convert=FALSE)
#Set working directory
setwd(gsub(pattern='Documents', replacement='Google Drive/Github/BikeSharingDataset', x=getwd()))

#Set seed
set.seed(101)

#Read data
TrainDF=read_delim(file=unz(description='Data/bike-sharing-demand.zip', filename='train.csv'), delim=',')

#Subset
X_Train=TrainDF[2000:nrow(TrainDF),c('temp', 'atemp', 'humidity', 'windspeed', 'casual', 'registered')]
Y_Train=as.matrix(TrainDF[2000:nrow(TrainDF),c('count')])
X_Test=TrainDF[1:1999,c('temp', 'atemp', 'humidity', 'windspeed', 'casual', 'registered')]
Y_Test=as.matrix(TrainDF[1:1999,c('count')])


#XVal
XVal=np$array(object=X_Train, dtype=np$float32)
XVal=np$reshape(XVal, c(8887L,2L,3L,1L))
#YVal
YVal=np$array(object=Y_Train)

#Keras Model
model=keras_model_sequential() 
model %>% 
  layer_conv_2d(filters=10, kernel_size=c(2,2), padding='same', activation='relu') %>%
  layer_conv_2d(filters=15, kernel_size=c(2,2), padding='same', activation='relu') %>%
  layer_conv_2d(filters=20, kernel_size=c(3,3), padding='same') %>%
  layer_max_pooling_2d(pool_size=c(2,2), strides=1) %>%
  layer_flatten() %>%
  layer_dense(units=30, activation='relu') %>%
  layer_dense(units=20, activation='relu') %>%
  layer_dense(units=10, activation='relu') %>%
  layer_dense(units=1)

#Compile model
model %>% compile(
  loss = 'mse',
  optimizer = optimizer_adam(),
  metrics = c('accuracy'))

#Train model

model %>% fit(XVal, YVal, epochs = 100)

Results=ldply(.data=Results, .fun=as.data.frame)
Results=cbind(CombinationGrid, Results)
write.table(x=Results, file='C:/Users/Tobal/Google Drive/Github/RKerasResult.txt', quote=FALSE, sep='\t', row.names=FALSE)

sessionInfo()

